akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/PathDirectivesExamplesSpec.scala

```diff
@@ -268,4 +269,100 @@ class PathDirectivesExamplesSpec extends RoutingSpec {
       handled shouldEqual false
     }
   }
+
+  "redirectToTrailingSlashIfMissing-0" in {
+    import akka.http.scaladsl.model.StatusCodes
+
+    val route =
+      redirectToTrailingSlashIfMissing(StatusCodes.MovedPermanently) {
+        path("foo"./) {
+          // We require the explicit trailing slash in the path
+          complete("OK")
+        } ~
+          path("bad-1") {
+            // MISTAKE!
+            // Missing `/` in path, causes this path to never match,
+            // because it is inside a `redirectToTrailingSlashIfMissing`
+            ???
+          } ~
+          path("bad-2/") {
+            // MISTAKE!
+            // / should be explicit as path element separator and not *in* the path element
+            // So it should be: "bad-1" /
+            ???
+          }
+      }
+
+    // Redirected:
+    Get("/foo") ~> route ~> check {
+      status shouldEqual StatusCodes.MovedPermanently
+
+      // results in nice human readable message,
+      // in case the redirect can't be followed automatically:
+      responseAs[String] shouldEqual {
+        "This and all future requests should be directed to " +
+          "<a href=\"http://example.com/foo/\">this URI</a>."
```

Comment:
This should actually say 'This and all future requests should be directed to the given URI.' (information exposure otherwise). But that's another issue.